### DISCUSSION TITLE
Route Caching, am I doing it wrong? or it can't handle this simple scenario?

### DISCUSSION THREAD

**Discussion by vhuerta** [vercel/next.js#60911] [App Router] [answered]
I was exploring Next.js for the first time, and within just 10 minutes, I noticed a significant issue with route caching – a feature that, in my opinion, is more of a drawback.

Consider the following simple app scenario:

**Route A:** A component (RSC) fetches records from Prisma and displays them in a list. Another component (RCC) has a button that triggers a server action that inserts a new record into the database and calls revalidatePath('/').

**Route B:** Contains a basic heading (h1) saying 'Hello, World.'

**The app layout consists of two Links, one pointing to Route A and the other to Route B. Both links have the prefetch={false} prop.**

Let's walk through the issues:

**Step 1:** Using Chrome, I navigate to Route A, click the button, and a new record appears in the list – so far, so good.

**Step 2:** Switching to Firefox, I navigate to Route A. Despite seeing one record initially, clicking the button adds another record to the list. Now, there are two records.

**Step 3:** Returning to Chrome, I still see only one record. Understandable, as it's not a real-time app. I navigate to Route B and then back to Route A. Surprisingly, I still see only one record. Clicking the button again updates the UI, now showing three records.

**Step 4:** Back in Firefox, I observe two records. Following the Route B and Route A navigation, the count remains at two. Clicking the button reloads the list, and now there are four records.

I can envision users encountering frustration quickly in a scenario like this during a Zoom call. Person A claims to have created the record, while person B, having navigated to a different page and back, doesn't see it. Person B decides to create it, leading to a duplicate entry. This user experience can be problematic.

I know this cache is only 30 seconds but imagine that the scenario I'm experiencing is not acceptable, can it be solved in some way with RSC without having to create an API endpoint and fetch the data from the client each time the component mounts? also, I don't think hard navigation is an acceptable solution

**Comment by SidKH** [comment, marked as answer]
Hey 👋
Here is an in-depth caching discussion with a proposal to make a 30-second cache configurable.

https://github.com/vercel/next.js/discussions/54075#discussioncomment-6754339
